ui/lib/kmip/addon/templates/components/edit-form-kmip-role.hbs

```diff
@@ -0,0 +1,80 @@
+{{#if model.canDelete}}
+  <Toolbar>
+    <ToolbarActions>
+      <ConfirmAction
+        @buttonClasses="toolbar-link"
+        @onConfirmAction={{action (perform save model (hash method="destroyRecord"))}}
+        @confirmMessage="This config will be permanently deleted."
+        data-test-edit-delete="true"
+      >
+        <span data-test-edit-delete-text>
+          {{deleteButtonText}}
+        </span>
+      </ConfirmAction>
+    </ToolbarActions>
+  </Toolbar>
+{{/if}}
+<form {{action (queue (action "preSave" model) (perform save model)) on="submit"}}>
```

Comment:
just so i understand the order of what's happening, is the lifecycle: 

1. user selects something in the form --> `updateModel`
1. user clicks save --> `preSave` ensures that if `operationNone` is true, then `operationAll` is false and then triggers -->
1. `save` --> the adapter's `serialize` method ensures that if `operationNone` is true, only `operationNone` is returned

Reply:
Yep, that's basically it, a couple of small clarifications: 
1. Yes, but there's also FormField which is toggling the state of the model attrs if you're clicking checkboxes in under "choose".
2. Yes, but `preSave` doesn't trigger the save `queue` calls them sequentially